js/tuxexchange.js

```diff
@@ -0,0 +1,533 @@
+        const tickerIds = Object.keys (tickersData);
+        const result = {};
+        for (let i = 0; i < tickerIds.length; i++) {
+            const id = tickerIds[i];
+            const tickerResult = this.parseTicker (id, tickersData[id]);
+            const symbol = tickerResult['symbol'];
+            result[symbol] = tickerResult;
+        }
+        return result;
+    }
+
+    parseTicker (marketId, tickerData) {
+        const timestamp = this.milliseconds ();
+        const market = this.findMarket (marketId);
+        return {
+            'symbol': market['symbol'],
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'close': this.safeFloat (tickerData, 'last'),
+            'last': this.safeFloat (tickerData, 'last'),
+            'high': this.safeFloat (tickerData, 'high24hr'),
+            'low': this.safeFloat (tickerData, 'low24hr'),
+            'percentage': this.safeFloat (tickerData, 'percentChange'),
+            'baseVolume': this.safeFloat (tickerData, 'baseVolume'),
+            'quoteVolume': this.safeFloat (tickerData, 'quoteVolume'),
+            'bid': this.safeFloat (tickerData, 'highestBid'),
+            'ask': this.safeFloat (tickerData, 'lowestAsk'),
+            'bidVolume': undefined,
+            'askVolume': undefined,
+            'vwap': undefined,
+            'open': undefined,
+            'previousClose': undefined,
+            'change': undefined,
+            'average': undefined,
+            'info': tickerData,
+        };
+    }
+
+    async fetchOrderBook (symbol, limit = undefined, params = {}) {
+        if (symbol === undefined) {
+            throw new ArgumentsRequired (this.id + ' fetchOrderBook () requires a symbol argument');
+        }
+        await this.loadMarkets ();
+        const codes = this.getIdsFromSymbol (symbol);
+        if (codes['base'] !== 'BTC') {
+            throw new NotSupported (this.id + ' this exchange only trades on symbols with BTC as base');
+        }
+        if (limit !== undefined) {
+            // At some point it would be worthwhile to extend the base parseOrderBook to include a limit
+            throw new NotSupported (this.id + ' fetchOrderBook () does not support a "limit" argument for this exchange');
+        }
+        let orderBookRequest = {
+            'coin': this.currencyId (codes['quote']),
+            'method': 'getorders',
+        };
+        const orderBook = await this.publicGet (this.extend (orderBookRequest, params));
+        const result = this.parseOrderBook (orderBook);
+        return result;
+    }
+
+    parseTrade (trade, market) {
+        // The API exposees different fields for the gettrades and getmytradehistory endpoints
+        const isPrivateTrade = trade['orderId'] !== undefined;
+        const tradeDate = this.parseDate (this.safeString (trade, 'date'));
+        const timestamp = tradeDate.valueOf ();
```

Comment:
We can't use `.valueOf` here, because it's JS-only. Instead of using parseDate, see how `timestamps` are parsed here: https://github.com/ccxt/ccxt/blob/master/js/binance.js#L932